app/helpers/task_list_helper.rb

```diff
@@ -6,8 +6,8 @@ def row_helper(trainee, task)
       publish_course_details:
         {
           task_name: "Course details",
-          path: edit_trainee_publish_course_details_path(trainee),
-          confirm_path: trainee_course_details_confirm_path(trainee),
+          path: (trainee.apply_application? ? trainee_apply_registrations_course_details_path(trainee) : edit_trainee_publish_course_details_path(trainee)),
+          confirm_path: (trainee.apply_application? ? trainee_apply_registrations_course_details_path(trainee) : trainee_course_details_confirm_path(trainee)),
```

Comment:
You might be wondering why the `confirm_path` goes to the apply course details page instead of the apply confirm page. Since a course can have multiple specialisms, we need to check for that first and this what the `Trainees::ApplyRegistrations::CourseDetailsController` does. If the course already has assigned subjects or only one subject specialism, we redirect the user to the confirm page. But if they have multiple specialisms to choose from, then the course details page will be render.